src/wasm/wasm-ir-builder.cpp

```diff
@@ -55,125 +47,141 @@ Result<Index> IRBuilder::addScratchLocal(Type type) {
   return Builder::addVar(func, name, type);
 }
 
-Result<> IRBuilder::push(Expression* expr) {
-  auto& exprStack = getExprStack();
+MaybeResult<IRBuilder::HoistedVal> IRBuilder::hoistLastValue() {
+  auto& stack = getScope().exprStack;
+  int index = stack.size() - 1;
+  for (; index >= 0; --index) {
+    if (stack[index]->type != Type::none) {
+      break;
+    }
+  }
+  if (index < 0) {
+    // There is no value-producing expression.
```

Comment:
Looks like an unreachable is counted as "value-producing", is that intentional? (seems so from the the later code, but maybe the comment can be improved here?)

Reply:
Yes, this is intentional. We should never treat the stack as being empty if there is an unreachable expression. I've clarified the comment to explicitly include unreachable instructions as well.